js/app.js

```diff
@@ -23,31 +23,14 @@ function initMap() {
     [-8189407.583341518, 5700582.732404122],  // Montreal
     [-7421303.096867932, 5772340.302359437],  // Fredericton
     [-7081032.809360132, 5569718.493209667]  // Halifax
-    //[-5867338.665139229, 6035201.501836645]   // St, John's
+    [-5867338.665139229, 6035201.501836645]   // St, John's
   ];
   map = new ol.Map({
     layers: [
       new ol.layer.Tile({
-        source: new ol.source.WMTS({
-           attributions: [
-            new ol.Attribution({
-              html:'<a href="http://www.openstreetmap.org/">OpenStreetMap</a>'
-            }),
-            ol.source.OSM.DATA_ATTRIBUTION
-          ],
-          url: 'http://demo.opengeo.org/geoserver/gwc/service/wmts/',
-          layer: 'osm:osm',
-          matrixSet: 'EPSG:900913',
-          format: 'image/png',
-          projection: projection,
-          tileGrid: new ol.tilegrid.WMTS({
-            origin: ol.extent.getTopLeft(projectionExtent),
-            resolutions: resolutions,
-            matrixIds: matrixIds
-          }),
-          style: '_null',
-          extent: [-16643092.788385,4534550.2653588,-5988382.5431409,11207197.085613]
-        })
+         source: new ol.source.XYZ({
+           url: 'http://api.boundlessgeo.io/v1/basemaps/boundless/osm/{z}/{x}/{-y}.png'
+         })
```

Comment:
You removed a whole bunch of options here, some of which "ol.source.XYZ" still uses. In particular: projection, tilegrid, attributions.
See here for more details: http://openlayers.org/en/v3.8.2/apidoc/ol.source.XYZ.html